File: Lab-1/Example1/ex1.cpp
#include <iostream>
#include <thread>

void printRange(int start, int end, int threadNum) {
    for(int i = start; i<=end; ++i){
        std::cout << "Thread " << threadNum << ':' << i << std::endl;
    }


}

int main(){

    std::thread t1(printRange,1,5,1);
    std::thread t2(printRange,6,10,2);

    t1.join();
    t2.join();

    std::cout <<"Both threads are finished !" << std::endl;

    return 0;
    // to compile this file run g++ -std=c++20 -O2 -pthread ex1.cpp -o ex1
}

File: Lab-1/Example1/ex1.java
 

public class ex1 {


    static class printRange extends Thread{

        int start;
        int end;
        int threadNumber;

        public printRange(int start,int end,int threadNumber){
            this.start = start;
            this.end = end;
            this.threadNumber = threadNumber;
        }

        public void run(){
            for(int i = start ; i <= end;i++ ){
                System.out.println("Thread "+threadNumber + ": " + i);
            }
        }
    }

    public static void main(String[] args) throws InterruptedException{

        printRange thread1 = new printRange(1,5,1);
        printRange thread2 = new printRange(6,10,2);
        
        thread1.start();
        thread2.start();
        
        thread1.join();
        thread2.join();

        System.out.println("Both threads finished.");

    }


}